components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/internal/UserStoreConfigComponent.java

```diff
@@ -97,6 +102,22 @@ protected void activate(ComponentContext ctxt) {
             log.debug("Identity User Store bundle is activated.");
         }
         try {
+            BundleContext bundleContext = ctxt.getBundleContext();
+            UserStoreConfigListenersHolder.getInstance().setBundleContext(bundleContext);
```

Comment:
We should not need to access BundleContext anywhere outside the "Component" class.